src/graph/util/ParserUtil.cpp

```diff
@@ -14,21 +14,21 @@ namespace graph {
 void ParserUtil::rewriteLC(QueryContext *qctx,
                            ListComprehensionExpression *lc,
                            const std::string &oldVarName) {
-  const auto &newVarName = qctx->vctx()->anonVarGen()->getVar();
-  qctx->ectx()->setValue(newVarName, Value());
+  qctx->vctx()->anonVarGen()->createVar(oldVarName);
+  qctx->ectx()->setValue(oldVarName, Value());
```

Comment:
Can you explain this?

Reply:
In List Comprehesion `[i in [...] | ...]`, the local defined variable `i` will be used in sub-expr, so I keep its origin name.

Reply:
What if the local var name same with the global var?

Reply:
Shadow global variable.